api/include/opentelemetry/trace/propagation/detail/context.h

```diff
@@ -9,15 +10,22 @@ namespace propagation
 namespace detail
 {
 
-inline trace::SpanContext GetCurrentSpan(const context::Context &context)
+inline nostd::shared_ptr<trace::Span> GetSpanFromContext(const context::Context &context)
 {
   context::ContextValue span = context.GetValue(trace::kSpanKey);
   if (nostd::holds_alternative<nostd::shared_ptr<trace::Span>>(span))
   {
-    return nostd::get<nostd::shared_ptr<trace::Span>>(span).get()->GetContext();
+    return nostd::get<nostd::shared_ptr<trace::Span>>(span);
   }
+  static nostd::shared_ptr<Span> invalid_span{new DefaultSpan(SpanContext::GetInvalid())};
```

Comment:
Question / concerns about the combination of `inline` and `magic static` here:

- magic statics may present a challenge on app termination, i.e. if someone tries to call into `GetSpanFromContext` in `std::atexit` handler. You'd most likely experience a crash.

- since you are returning a `shared_ptr` to what appears to be `invalid_span`, what prevents me to overwrite the contents of that ptr with my own span then. And the next caller then would actually get my span, not the invalid span, i.e. appears like previously the code was safe. Now I can hijack and override the invalid span with my own span.

Can you double-check on this?

Reply:
Both are valid points. Thanks for noticing it. I don't think we should be returning static in case of invalid_span. We also don't do it here: 

https://github.com/open-telemetry/opentelemetry-cpp/blob/c7f49debf3331ef912a8a20e0b1f29b0a67317ae/api/include/opentelemetry/trace/tracer.h#L162

Will fix it accordingly.

Reply:
I checked that I might have been wrong about what'd happen. With clang c++11 appears like my 2nd concern is maybe wrong.. But most likely you can simply remove the `static` keyword at line 20. This would be nicer to read. I like your last suggestion!